### PR TITLE
[Table] Partially fix issues with scroll while drag-selecting

### LINKED ISSUE #950: Multi-row selection cut off during scrolling
When selecting more than what's visible at a time by scrolling during selection, the selection is cut off at the bottom to what's currently visible, see gif:
bluprint-table-bug

### PR DESCRIPTION
**Pull request by cmslewis**
### ❗  DO NOT MERGE (let's merge #1114 first while I investigate auto-scrolling more) 
---

#### Checklist

- [x] Include tests

#### Changes proposed in this pull request:

- Fix issues with manual scrolling while a drag-selection is in progress: 
    - Mousing down on a **row** header and scrolling anchors the selection to that original row.
    - Mousing down on a **column** header and scrolling anchors the selection to that original column.
    - Mousing down on a **cell** and scrolling anchors the selection that original cell.
- Mousing down on a **cell** and dragging past the edge of the table will auto-scroll the table for you, expanding the selection in the process.

NOTE: I'm not addressing #505 here after all, because we observed some buggy behavior that I couldn't immediate figure out how to solve: clicking and column header and dragging to scroll the table would stop expanding the selection after ~10 columns or so and prevent you from adjusting the drag-selection further. Leaving #505 open to track that.

#### Reviewers should focus on:

- Is it bad to not return -1 from the affected `Locator` functions anymore? I couldn't see a case when we'd really need that behavior anymore.

#### Screenshot

Scrolling while drag-selecting cells - works great:

### CODE REVIEW

**Comment by llorca** [comment]
Summary from the discussion offline: we're reinventing click+drag scrolling, even though it's already supported natively in the browser. We have handlers on headers that prevent native scrolling from triggering—we should find the root cause and investigate leveraging native scroll as much as possible. I'm pretty sure it's related to this other issue about not being able to scroll when hovering headers: https://github.com/palantir/blueprint/issues/957

Aside from that, there are some nice improvements in this PR that can be merged separately such as selection with scroll at the same time correctly updating the selection region.

File: gulp/util/karma-config.js [hunk 1]
```diff
@@ -40,8 +40,8 @@ module.exports = function createConfig(project) {
         coverageReporter: {
             check: {
                 each: {
-                    lines: 79,
-                    statements: 79,
+                    lines: 0, // TODO: Undo before merging.
```
Comment (gscshoyru) [inline gulp/util/karma-config.js]: Don't forget :)

File: packages/table/src/headers/columnHeader.tsx [hunk 2]
```diff
@@ -271,6 +275,7 @@ export class ColumnHeader extends React.Component<IColumnHeaderProps, IColumnHea
     }
 
     private handleDragSelectableSelectionEnd = () => {
+        this.activationCol = null;
```
Comment (gscshoyru) [inline packages/table/src/headers/columnHeader.tsx]: I realize this is probably good practice and we should keep it -- just asking from a code understanding perspective -- is this necessary? You're setting this on start of drag every time, I'm just making sure that if you _wanted_ to remove this line you could.

File: packages/table/src/locator.ts [hunk 3]
```diff
@@ -24,24 +24,22 @@ export interface ILocator {
...
-     * the coordinate is not over a row.
+     * Locates a row's index given the client Y coordinate.
      * If `useMidpoint` is `true`, returns the index of the row whose top
      * edge is closest, splitting on the midpoint of each row.
      */
     convertPointToRow: (clientY: number, useMidpoint?: boolean) => number;
 
     /**
      * Locates a cell's row and column index given the client X
-     * coordinate. Returns -1 if the coordinate is not over a table cell.
```
Comment (gscshoyru) [inline packages/table/src/locator.ts]: Why?

**Comment by gscshoyru** [comment]
@llorca I'm not sure what you're saying re: clicking and dragging being native to browsers is true. It's only true insofar as you're highlighting things. I'm pretty sure we had to reinvent click and drag with slate as well when moving widgets; it won't work natively if you override drag behavior (as we must)

File: packages/table/src/headers/columnHeader.tsx (hunk 2 above)
Comment (cmslewis) [inline packages/table/src/headers/columnHeader.tsx]: No, not strictly necessary. :)

File: packages/table/src/locator.ts (hunk 3 above)
Comment (cmslewis) [inline packages/table/src/locator.ts]: If you _don't_ return -1 from here—and instead return the next index beyond the index currently in view—then the viewport will move to out-of-view indices as desired. If you return -1, the viewport doesn't update, making the whole drag-to-scroll feature stop working.

**Comment by llorca** [comment]
@gscshoyru check out this GIF, first scroll try is what this PR is about, the scrolling stops intermittently and doesn't accelerate the further I move the mouse. The second one (when I start from column BV), is a much better experience, it's the native interaction. Only problem is that it doesn't extend the selection as you can see:

screen recording google chrome

**Comment by llorca** [comment]
Just checked this out again. The GIF from my previous is still accurate, the current scrolling experience in the body is subpar -- feels like a bandaid: mouse needs to move at all times, no inertia 👎 

There seems to be a more pressing concern though, it looks like we broke something around column selection. Check out the new GIF attached: I can do all sorts of cool selection, but when I select a column, selection is frozen and I can't create a new region anymore.

screen recording google chrome

**Comment by cmslewis** [comment]
@llorca will look into it

**Comment by cmslewis** [comment]
@llorca @gscshoyru I pulled out the direct fix to #950 into its own PR. Let's merge that first while I look more into the auto-scrolling behavior.

**Comment by cmslewis** [comment]
This is addressed by `Shift + Click`. Will pick this particular feature up later.

**Pull request closed without merging**